packages/mini-browser/src/browser/mini-browser-open-handler.ts

```diff
@@ -22,7 +23,19 @@ export interface MiniBrowserOpenerOptions extends WidgetOpenerOptions, MiniBrows
 }
 
 @injectable()
-export class MiniBrowserOpenHandler extends WidgetOpenHandler<MiniBrowser> {
+export class MiniBrowserOpenHandler extends WidgetOpenHandler<MiniBrowser> implements FrontendApplicationContribution {
+
+    /**
+     * This is a **hack**.
```

Comment:
Why a hack? Isn't it more like an optimization (caching)?
I would prefer not to back such a big deal about it, since it doesn't really look odd.
This sort of comments will leave the impression that it has to be fixed somehow.

Reply:
Well, do original idea did not include any caching but it did not work [due to the `sync`/`async` issue with the Phosphor context menu](https://github.com/theia-ide/theia/issues/1959). This change was introduced to work around that problem, but you're right, perhaps I could get rid of the `This is a hack` introduction.